### PR TITLE
[SYCL][NFC] Code cleanup (phase 2) revealed by self build.

### PR DESCRIPTION
**Pull request by zahiraam**

### CODE REVIEW

File: llvm/lib/ExecutionEngine/IntelJITEvents/jitprofiling.c [hunk 1]
```diff
@@ -19,7 +19,9 @@
 
 #if ITT_PLATFORM==ITT_PLATFORM_WIN
 #include <windows.h>
+#if !INTEL_CUSTOMIZATION
 #pragma optimize("", off)
+#endif // !INTEL_CUSTOMIZATION
```
Comment (bader) [inline llvm/lib/ExecutionEngine/IntelJITEvents/jitprofiling.c:24]: I'm not sure what this does, but I suggest committing this change to llorg instead.

**Review by mdtoguchi — APPROVED** [review]
OK for driver

File: llvm/lib/ExecutionEngine/IntelJITEvents/jitprofiling.c (hunk 1 above)
Comment (zahiraam) [reply llvm/lib/ExecutionEngine/IntelJITEvents/jitprofiling.c]: Done.

**Review by bader — APPROVED** [review]
llvm/* changes looks okay to me.

**Review by premanandrao — APPROVED** [review]
SemaSYCL change looks good to me.

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**